Samples/NetworkMediaStreamer/CameraRTPStreamerApp/CameraRTPStreamerApp.cpp

```diff
@@ -0,0 +1,419 @@
+        {
+            if (fs.Value().Controller().VideoDeviceController().Id() == s.VideoDeviceId())
+            {
+                selectedFs = fs.Value();
+            }
+        }
+        auto fr = mc.CreateFrameReaderAsync(selectedFs, selectedFs.CurrentFormat().Subtype(), sz).get();
+        fr.AcquisitionMode(MediaFrameReaderAcquisitionMode::Realtime);
+        slim_mutex m;
+        std::vector<com_ptr<IMFSinkWriter>> spSinkWriters;
+        auto strmIter = streamers.First();
+        for (uint32_t i = 0; i < streamers.Size(); i++)
+        {
+            winrt::com_ptr<IMFSinkWriter> spSW;
+            spSW.attach(InitSinkWriter(strmIter.Current().Value().as<IMFMediaSink>().get(), selectedFs.CurrentFormat()));
+            spSinkWriters.push_back(spSW);
+            strmIter.MoveNext();
+        }
+        fr.FrameArrived([&](MediaFrameReader mfr, MediaFrameArrivedEventArgs args)
+            {
+                slim_lock_guard g(m);
+                auto mf = mfr.TryAcquireLatestFrame();
+                if (!mf) return;
+                auto vmf = mf.VideoMediaFrame();
+                if (!vmf) return;
+                auto vf = vmf.GetVideoFrame();
+                if (!vf) return;
+                winrt::com_ptr<IMFSample> spSample;
+                vf.as<IVideoFrameNative>()->GetData(__uuidof(IMFSample), spSample.put_void());
+                for (uint32_t i = 0; i < spSinkWriters.size(); i++)
+                {
+                    spSinkWriters[i]->WriteSample(0, spSample.get());
+                }
+            });
+        fr.StartAsync();
+#else
+
+        auto me = MediaEncodingProfile::CreateMp4(VideoEncodingQuality::Auto);
+        auto lowLagRec = mc.PrepareLowLagRecordToCustomSinkAsync(me, streamers.First().Current().Value().as<IMediaExtension>()).get();
+        lowLagRec.StartAsync().get();
+#endif
+
+        std::cout << "\nStarted Capture and RTSP Server.\n";
+        auto hostnames = winrt::Windows::Networking::Connectivity::NetworkInformation::GetHostNames();
+        std::cout << "\nAvailable links:\n";
+        for (auto hname : hostnames)
+        {
+            for (auto s : streamers)
+            {
+                std::wcout << L"rtsp://" << hname.DisplayName().c_str() << L":" << ServerPort << s.Key().c_str() << std::endl;
+                if (serverHandleSecure)
+                {
+                    std::wcout << L"rtsps://" << hname.DisplayName().c_str() << L":" << SecureServerPort << s.Key().c_str() << std::endl;
+                }
+            }
+        }
+
+        char c = 1;
+        while (c != 0)
+        {
+            std::cout << "\n 1. Add User\n 2. Remove User \n 0. Quit\n You choice:" ;
+
+            std::cin >> c;
+            switch (c)
+            {
+            case '1':
+            {
+                std::string username, password;
+                std::cout << "\nEnter username :-  ";
+                std::cin >> username;
+                std::cout << "\nEnter new password :- ";

```

Comment:
>Enter new password [](start = 32, length = 18)

If this is really a new password do we want a confirmation prompt to make sure they entered the password as they expected?

Reply:
ideally yes - in an actual product, i would like to hide the password as well.... However, this is just a test application, not sure if we need to make it complicated.. let me know what you think..

---
In reply to: [491139424](https://github.com/microsoft/Windows-Camera/pull/13#discussion_r491139424) [](ancestors = 491139424)